pkg/telemetry/storage/db_test.go

```diff
@@ -148,13 +149,97 @@ func TestCollectAndFetchAndPublish(t *testing.T) {
 		p.Start()
 
 		require.Eventually(t, func() bool {
-			metrics, err = s.FetchUnpublishedMetrics(context.Background(), 2)
+			metrics, err = s.FetchMetrics(context.Background(), false, 2)
+			require.NoError(t, err)
+			return len(metrics) == 0
+		}, 5*time.Second, time.Second)
+
+		p.Close()
+	})
+
+	t.Run("read query", func(t *testing.T) {
+		// collect two mocked read query metrics
+		readQueryMetrics := [2]readQuery{{}, {}}
+		require.NoError(t, telemetry.Collect(context.Background(), readQueryMetrics[0]))
+		require.NoError(t, telemetry.Collect(context.Background(), readQueryMetrics[1]))
+
+		metrics, err := s.FetchMetrics(context.Background(), false, 10)
+		require.NoError(t, err)
+		require.Len(t, metrics, 2)
+
+		for i, metric := range metrics {
+			require.Equal(t, telemetry.ReadQueryType, metric.Type)
+			require.Equal(t, 1, metric.Version)
+			require.False(t, metric.Timestamp.IsZero())
+
+			payload := metric.Payload.(*telemetry.ReadQueryMetric)
+			require.Equal(t, readQueryMetrics[i].IPAddress(), payload.IPAddress)
+			require.Equal(t, readQueryMetrics[i].SQLStatement(), payload.SQLStatement)
+			require.Equal(t, readQueryMetrics[i].Unwrap(), payload.Unwrap)
+			require.Equal(t, readQueryMetrics[i].Extract(), payload.Extract)
+			require.Equal(t, readQueryMetrics[i].Output(), payload.Output)
+		}
+
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+		}))
+		defer ts.Close()
+
+		exporter, err := publisher.NewHTTPExporter(ts.URL, "")
+		require.NoError(t, err)
+		nodeID := strings.Replace(uuid.NewString(), "-", "", -1)
+		p := publisher.NewPublisher(s, exporter, nodeID, time.Second)
+		p.Start()
+
+		require.Eventually(t, func() bool {
+			metrics, err = s.FetchMetrics(context.Background(), false, 2)
 			require.NoError(t, err)
 			return len(metrics) == 0
 		}, 5*time.Second, time.Second)
 
 		p.Close()
 	})
+
+	t.Run("delete old metrics", func(t *testing.T) {
+		// clear store
```

Comment:
Also, kind of regretting to have made store a global variable in the pkg. It makes things a lot easier for clients but a lot harder to test. I have to clear the store so this test runs correctly because the store has data from previous tests.

Reply:
Yeah, I remember this pain when I wanted to parallelize the tests.